docs/007-funding-rate.adoc

```diff
@@ -0,0 +1,96 @@
+No funding rate needs to paid for the period of time _before_ the first renewal.
+
+== Context
+
+The funding rate is the cost of holding a perpetual swap position that is on the side of the market sentiment.
+There needs to be a funding rate because that is what motivates traders to keep their perpetual swaps positions open _against_ the market sentiment.
+The sign of the funding rate determines which of the two parties involved must pay the funding fee to keep their position open.
+
+In the 10101 ecosystem, the funding rate must come from the 10101 orderbook.
+This value will be updated at a fixed interval e.g. every 8 hours.
+
+== Options considered
+
+=== Lightning payment before the DLC is renewed
+
+The renewal protocol will first require one of the two parties (coordinator and trader) to generate an invoice for the funding fee.
+Which party generates the invoice (i.e. the receiver) is determined by the sign of the funding rate of the 10101 orderbook.
+
+Only after the invoice has been paid will the coordinator initiate the renew protocol to roll over the DLC between coordinator and trader.
+
+==== Pros
+
+- The DLC remains unaffected by the exchanged funding rate.
+- Can be implemented without changes to our dependencies.
+- Easy to track the funding fee payments.
+- Fee is readily available to use after.
+
+==== Cons
+
+===== No execution guarantees
+
+There is no guarantee that the renew protocol will succeed after the funding fee payment is claimed.
+Depending on the size of the position, it could be economically beneficial for a party to claim the funding fee paid by the other party and then force-close the channel.
+As such, in some scenarios we would be trusting the counterparty to not run away with an unearned funding fee.
+
+With PTLCs we might be able to make the DLC channel update and the payment atomic, but that is not yet implemented.
+
+===== Funds required on Lightning channel
+
+To be able to pay the funding fee the payer will need to have sufficient outbound liquidity.
+It might happen that a user will need to stop trading if their Lightning channel cannot cover the costs.
+
+They would still be able to close their position and open a new one.
+
+=== Adjust the CET payouts depending on the funding fee
+
+If the funding fee is `X` paid from `A` to `B`, for every `o_A`, CET output amount belonging to `A`, we would subtract `X`: `o_A - X`; and for every `o_B`, CET output belonging to `B`, we would add `X`: `o_B + X`.
+
+Unfortunately this doesn't work in all scenarios:
+
+- If a CET already only pays to `B` then the payout cannot be increased!
+- If a CET has `o_A < X`, then it will not be possible to increase `o_B` sufficiently.
+
+As such, in some scenarios the fee will not be paid.
+This problem might compound as the same position might be rolled over multiple times.
+If the advantaged party ends up getting liquidated, the winning party will not see any funding fee.
```

Comment:
Imho this is a good design: the longer the position is open, the earlier the user gets liquidated because its margin is being reduced by the funding rate. 

Expressed differently: 

- Alice has margin `m_A`
- Bob has margin `m_B`
- Funding rate is `f`.

After the first rollover, e.g. after 8h, we have 
- Alice margin `m_A - f`
- Bob margin `m_B + f`

The liquidation for the party going long is calculated as: 
`= (price_opening * leverage)/(leverage+1)`
 
Now, the important thing is that the `leverage` is only constant at the beginning when the position was opened, later the `leverage` is a function of the `margin` and the `price`, i.e. 
- initially we had: `margin = quantity / (price_opening * leverage)`
- after the opening we have `leverage = quantity / (price_opening * margin)`

Which means, the liquidation price will change but each party will always get the funding rate until either of the party gets liquidated, i.e. does not have enough money to pay for the funding rate.

Reply:
There are a couple of reasons why I don't like this idea so much:

- Makes liquidation more likely, which is the worst case scenario for traders and is expensive and cumbersome to enforce.
- Traders can miss out on funding fees (see [comment above](https://github.com/get10101/10101/pull/1233#discussion_r1317840973)).

Reply:
> Makes liquidation more likely, which is the worst case scenario for traders and is expensive and cumbersome to enforce.

Yes, but I don't think it's that much more likely, e.g. current funding rate on BitMEX is 0.01% for 8h, meaning, his margin gets reduced by 0.01% and it's 0.01% more likely that you get liquidated. That's like nothing :D